tests/FSharp.Compiler.ComponentTests/Language/ExtensionMethodTests.fs

```diff
@@ -114,6 +114,43 @@ namespace Consumer
 
         csharp |> compile |> shouldSucceed
         
+    [<Fact>]
+    let ``F# CSharpStyleExtensionMethod consumed in F#`` () =
+        let producer =
+            FSharp
+                """
+namespace Producer
+
+open System.Runtime.CompilerServices
+
+type WidgetBuilder<'msg, 'marker>() = class end
+
+type IMarkerOne = interface end
+
+// Commenting out [<Extension>] breaks
+//[<Extension>]
+type WidgetBuilderExtensions =
```

Comment:
The problem here is when unpickling the binary of the `Producer`, there is no attribute present on the type.

In https://github.com/dotnet/fsharp/pull/13839 we are pushing the attribute to the type, and that must somehow not get pickled.

Reply:
![image](https://github.com/dotnet/fsharp/assets/2621499/e5537b77-f8e2-4734-83f4-5832ea708de0)

Reply:
The signature data does not have the attribute, there lies the problem.